### PR TITLE
FTR-81 Fix modularity

### PR DESCRIPTION
**Pull request by mpuskadij**
List of changes and descriptions

- build.gradle.kts (app) - imported maps module (for interface IMapProvider) and implementations of IMapProvider located in modules _osm_ and _google_maps_
- IChosenMapProvider - data layer interface for future use if chosen map provider is not to be saved in SharedPreferences, but somewhere else
- SharedPreferencesManager - singleton, offers saving and retrieving strings to shared preferences, but also implements interface IChosenMapProvider
- IMapProviderStateManager - business logic layer interface, made for future use if serviceloader or another way of retrieving interface implementations is found
- MapProviderManager - implements IMapProviderStateManager, uses ServiceLoader java class to load implementations of  interface IMapProvider
- CompanyLocation - added a check if no provider exists, then it displays a message that no maps exist, otherwise gets the selected map and draws a map, added a clipToBounds to box as modifier so that osm doesn't break bounds
- MapDropdown - UI component that creates a exposable dropdown for map providers. If no map providers exist, menu is visually disabled
- CompanyDetailsContent - added a check if map provider list is empty so that "Location" header and box containing the map are not displayed
- ProfileContent - added a new info box displaying chosen map provider
- added 2 new modules - _osm_ and _google_maps_
- updated IMapProvider interface - new method getName is required for every implementation so that the dropdown can be populated and saved to shared preferences

Screenshot_20241226-140656_ServiceSync 1
Screenshot_20241226-135253_ServiceSync 1
Screenshot_20241226-132359_ServiceSync 1
Screenshot_20241226-143741_ServiceSync 1

### CODE REVIEW

File: app/src/main/java/hr/foi/air/servicesync/backend/IChosenMapProviderSaver.kt
Comment (ibeusan20) [inline app/src/main/java/hr/foi/air/servicesync/backend/IChosenMapProviderSaver.kt:1]: Complying to OOP, very nice!

File: app/src/main/java/hr/foi/air/servicesync/backend/SharedPreferencesManager.kt
Comment (ibeusan20) [inline app/src/main/java/hr/foi/air/servicesync/backend/SharedPreferencesManager.kt:1]: Nice job working with shared preferences!

File: app/src/main/java/hr/foi/air/servicesync/business/MapProviderManager.kt
Comment (ibeusan20) [inline app/src/main/java/hr/foi/air/servicesync/business/MapProviderManager.kt:1]: Very clean way of implementing this in an OOP compliant way, well done.

File: app/src/main/java/hr/foi/air/servicesync/ui/components/CompanyLocation.kt [hunk 1]
```diff
@@ -3,16 +3,32 @@ package hr.foi.air.servicesync.ui.components
...
+import hr.foi.air.servicesync.R
+import hr.foi.air.servicesync.business.MapProviderManager
 
 @Composable
-fun CompanyLocation(geoPoint: GeoPoint, mapProvider: IMapProvider)
+fun CompanyLocation(geoPoint: GeoPoint)
 {
-    Box(modifier = Modifier.fillMaxWidth().height(200.dp)) {
-        mapProvider.CreateMap(geoPoint.latitude,geoPoint.longitude)
+    val mapProvider = MapProviderManager.getCurrentMapProvider(context = LocalContext.current)
```
Comment (ibeusan20) [inline app/src/main/java/hr/foi/air/servicesync/ui/components/CompanyLocation.kt:20]: Nice usage of a new cool function. When there is no map providers there's no maps, makes sense, nice!

File: app/src/main/java/hr/foi/air/servicesync/ui/components/MapDropdown.kt
Comment (ibeusan20) [inline app/src/main/java/hr/foi/air/servicesync/ui/components/MapDropdown.kt:1]: Aldo this is a fairly specific component, well done making it reusable and placing it in it's own file.

File: app/src/main/java/hr/foi/air/servicesync/ui/contents/CompanyDetailsContent.kt
Comment (ibeusan20) [inline app/src/main/java/hr/foi/air/servicesync/ui/contents/CompanyDetailsContent.kt:1]: Well done using already mentioned "no providers - no maps" function.

File: app/src/main/java/hr/foi/air/servicesync/ui/contents/ProfileContent.kt [hunk 2]
```diff
@@ -116,6 +120,12 @@ fun ProfileContent(modifier: Modifier = Modifier) {
                     visualTransformation = PasswordVisualTransformation(),
                     modifier = Modifier.fillMaxWidth()
                 )
+                MapDropdown(
```
Comment (ibeusan20) [inline app/src/main/java/hr/foi/air/servicesync/ui/contents/ProfileContent.kt:123]: Placing map provides choice on the profile screen is a nice choice.

File: app/src/main/res/values/strings.xml
Comment (ibeusan20) [inline app/src/main/res/values/strings.xml:1]: Good practice!

**Review by ibeusan20 — APPROVED** [review]
Overall, well done!

File: app/src/main/java/hr/foi/air/servicesync/backend/SharedPreferencesManager.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package hr.foi.air.servicesync.backend
+
+import android.content.Context
+import androidx.preference.PreferenceManager
+
+object SharedPreferencesManager : IChosenMapProviderSaver {
```
Comment (dblazevic21) [inline app/src/main/java/hr/foi/air/servicesync/backend/SharedPreferencesManager.kt:6]: Decent usage of objects with overriden functions

**Review by dblazevic21 — APPROVED** [review]
Good improvement with the interfaces, added strings and objects

**Pull request merged**